fdbclient/admin_client.go

```diff
@@ -0,0 +1,592 @@
+/*
+ * admin_client.go
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fdbclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	fdbtypes "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta1"
+	controllers "github.com/FoundationDB/fdb-kubernetes-operator/controllers"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var adminClientMutex sync.Mutex
+
+var maxCommandOutput = parseMaxCommandOutput()
+
+func parseMaxCommandOutput() int {
+	flag := os.Getenv("MAX_FDB_CLI_OUTPUT_LENGTH")
+	if flag == "" {
+		return 20
+	}
+	result, err := strconv.Atoi(flag)
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+var exclusionLinePattern = regexp.MustCompile("(?m)^ +(.*)$")
+var protocolVersionRegex = regexp.MustCompile(`(?m)^protocol (\w+)$`)
+
+// cliAdminClient provides an implementation of the admin interface using the
+// FDB CLI.
+type cliAdminClient struct {
+	// Cluster is the reference to the cluster model.
+	Cluster *fdbtypes.FoundationDBCluster
+
+	// clusterFilePath is the path to the temp file containing the cluster file
+	// for this session.
+	clusterFilePath string
+}
+
+// NewCliAdminClient generates an Admin client for a cluster
+func NewCliAdminClient(cluster *fdbtypes.FoundationDBCluster, _ client.Client) (controllers.AdminClient, error) {
```

Comment:
I guess `_ client.Client` is only there for backwards compatibility?

Reply:
Yes